chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,187 @@
+import numpy
+import six
+
+from chainer import cuda
+from chainer import reporter
+from chainer import training
+from chainer.training import extension
+
+
+def _iterable(x):
+    if isinstance(x, (list, tuple)):
+        return x
+    return x,
+
+
+def _prefix_dict_keys(prefix, x):
+    return {'{}/{}'.format(prefix, k): v for k, v in six.iteritems(x)}
+
+
+def _target_name(link, param_names, attr_names):
+    """Generate a dictionary key based on context."""
+    link_name = getattr(link, 'name', 'None')
+    param_name = '-'.join(_iterable(param_names))
+    attr_name = '-'.join(_iterable(attr_names))
+
+    return '{}/{}/{}'.format(link_name, param_name, attr_name)
+
+
+def _get_link_params(link, param_names, attr_names):
```

Comment:
I feel you only need to support `str`s for `param_names` and `attr_names`. I mean, `param_name` and `attr_name` are enough.